athena-federation-integ-test/src/main/java/com/amazonaws/athena/connector/integ/providers/ConnectorVpcAttributesProvider.java

```diff
@@ -0,0 +1,133 @@
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.amazonaws.athena.connector.integ.providers;
+
+import com.amazonaws.athena.connector.integ.data.ConnectorVpcAttributes;
+import com.amazonaws.athena.connector.integ.data.ConnectorVpcSubnetAttributes;
+import com.amazonaws.services.ec2.AmazonEC2;
+import com.amazonaws.services.ec2.AmazonEC2ClientBuilder;
+import com.amazonaws.services.ec2.model.DescribeSecurityGroupsResult;
+import com.amazonaws.services.ec2.model.DescribeSubnetsResult;
+import com.amazonaws.services.ec2.model.DescribeVpcsResult;
+import com.amazonaws.services.ec2.model.SecurityGroup;
+import com.amazonaws.services.ec2.model.Subnet;
+import com.amazonaws.services.ec2.model.Vpc;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Responsible for providing the Connector's VPC attributes used in creating the Connector's stack attributes.
+ */
+public class ConnectorVpcAttributesProvider implements AutoCloseable
+{
+    private static final Logger logger = LoggerFactory.getLogger(ConnectorVpcAttributesProvider.class);
+
+    private final AmazonEC2 ec2Client;
+
+    public ConnectorVpcAttributesProvider()
+    {
+        this.ec2Client = AmazonEC2ClientBuilder.defaultClient();
+    }
+
+    /**
+     * Gets the default VPC configuration used for configuring the Lambda function.
+     * @return VPC attributes (VPC Id, Security group Id, Subnet Ids, and Availability zones) if a VPC configuration is
+     * supported for this connector.
+     * @throws RuntimeException Errors were encountered obtaining the VPC configuration.
+     */
+    public ConnectorVpcAttributes getAttributes()
+            throws RuntimeException
+    {
+        final String vpcId = getDefaultVpcId();
+        final String securityGroupId = getSecurityGroupId(vpcId);
+        final ConnectorVpcSubnetAttributes subnetAttributes = getSubnetAttributes(vpcId);
+
+        logger.info("VPC Id: [{}], SG: [{}], {}", vpcId, securityGroupId, subnetAttributes);
+
+        return new ConnectorVpcAttributes(vpcId, securityGroupId, subnetAttributes);
+    }
+
+    /**
+     * Gets the default VPC Id used for configuring the Lambda function (e.g. vpc-xxxx).
+     * @return VPC Id
+     * @throws RuntimeException No VPCs were found.
+     */
+    private String getDefaultVpcId()
+            throws RuntimeException
+    {
+        DescribeVpcsResult vpcsResult = ec2Client.describeVpcs();
+        for (Vpc vpc : vpcsResult.getVpcs()) {
+            if (vpc.getIsDefault()) {
```

Comment:
why the default VPC?

Reply:
will change logic in favor of config file.